src/ansys/mapdl/core/launcher.py

```diff
@@ -2109,8 +2119,10 @@ def get_version(
     if not version:
         version = os.getenv("PYMAPDL_MAPDL_VERSION")
 
-    if not version:
-        # Early exit
+    if not version and exec_file and _HAS_ATP:
+        version = version_from_path("mapdl", exec_file)
+
+    elif not version:
```

Comment:
A else could be enough here because you will return and not execute following lines.
```suggestion
    else not version:
```